app/controllers/balance_transactions_controller.rb

```diff
@@ -0,0 +1,73 @@
+class BalanceTransactionsController < ApplicationController
+  before_action :authenticate_user!
+
+  def new
+    @balance_transaction = BalanceTransaction.new
+    @balance_transaction.transactions.build
+  end
+
+  def create
+    byebug
+    respond_to do |format|
+      notice = BalanceTransactions::Creator.new(create_params).call ? t('.success') : t('.failure')
+      format.html { redirect_to transactions_path, notice: notice }
+      format.js { render layout: false }
+    end
+  end
+
+  def edit
+    @balance_transaction = transaction.transactinable
+  end
+
+  def update
+    byebug
+    respond_to do |format|
+      notice = BalanceTransactions::Updater.new(update_params).call ? t('.success') : t('.failure')
+      format.html { redirect_to transactions_path, notice: notice }
+      format.js { render layout: false }
+    end
+  end
+
+  def destroy
+    respond_to do |format|
+      notice = BalanceTransactions::Destroyer.new(balance_transaction_params[:id]) ? t('.delete_success') : t('.failure')
+      format.html { redirect_to transactions_path, notice: notice }
+      format.js { render layout: false }
+    end
+  end
+
+  private
+
+  def balance_transaction_params
+    params.require(:balance_transaction).permit(
+      :id,
+      :comment,
+      :date,
+      transactions: {}
+    )
+  end
+
+  def params_to_hash
```

Comment:
@sinpai  Let's move this method over `#balance_transaction_params`

Reply:
@marderer can you explain why should we move methods after a direct call from up to down?

Reply:
@AleksSenkou because we read from top to bottom